resources/js/Pages/Campaign/Detail/CampaignDetail.tsx

```diff
@@ -1,32 +1,94 @@
+import {Typography} from '@mui/material'
 import {ThemeProvider} from '@mui/material/styles'
 import useEditDrawer from '@Utils/use-edit-drawer'
-import {CampaignDetailBox, EntryTable} from 'Components'
+import {
+    CampaignCreateForm,
+    CampaignDetailBox,
+    Drawer,
+    EntryCreateForm,
+    EntryTable,
+} from 'Components'
 import {ApplicationLayout} from 'Layouts'
-import {entriesData} from 'Mock/entries-data'
-import React from 'react'
+import React, {useState} from 'react'
 import {CampaignData} from 'Types/campaign-data'
+import {CharacterData} from 'Types/character-data'
+import {EntriesData} from 'Types/entries-data'
+import {UserCharacterData} from 'Types/user-character-data'
 import {getFontTheme} from 'Utils'
 
 const theme = getFontTheme('Form', 14)
 
 type CampaignDetailPropType = {
     campaign: CampaignData
+    userCharacter: UserCharacterData
+    gameMasters: any[]
+    characters: CharacterData[]
 }
 
-const CampaignDetail = () => (
-    <ThemeProvider theme={theme}>
-        <CampaignDetailBox
-        // campaign={campaign}
-        />
-        <EntryTable
-            data={entriesData}
-            setEditEntryId={useEditDrawer}
-            setEditEntryData={useEditDrawer}
-            setIsEditDrawerOpen={useEditDrawer}
-        />
-    </ThemeProvider>
-)
-
+const CampaignDetail = ({
+    campaign,
+    userCharacter,
+    gameMasters,
+    characters,
+}: CampaignDetailPropType) => {
+    const [isEditEntryDrawerOpen, setIsEditEntryDrawerOpen] = useState<boolean>(false)
+    const [editEntryData, setEditEntryData] = useState<EntriesData>()
+    const [editEntryId, setEditEntryId] = useState<number>(0)
+    const {isEditDrawerOpen, setIsEditDrawerOpen} = useEditDrawer<CampaignData>()
```

Comment:
instead of these, use the hook for the entry drawer since it will require all the properties in the hook returned object

for the campaign edit drawer, you only need 1 `useState`